File: app/src/main/java/player/phonograph/helper/ShuffleHelper.java
package player.phonograph.helper;

import androidx.annotation.NonNull;

import player.phonograph.model.Song;

import java.util.Collections;
import java.util.List;

/**
 * @author Karim Abou Zeid (kabouzeid)
 */
public class ShuffleHelper {

    public static void makeShuffleList(@NonNull List<Song> listToShuffle, final int current) {
        if (listToShuffle.isEmpty()) return;
        if (current >= 0) {
            Song song = listToShuffle.remove(current);
            Collections.shuffle(listToShuffle);
            listToShuffle.add(0, song);
        } else {
            Collections.shuffle(listToShuffle);
        }
    }
}

File: app/src/main/java/player/phonograph/helper/ShuffleHelper.kt
package player.phonograph.helper

import player.phonograph.model.Song

/**
 * @author Karim Abou Zeid (kabouzeid)
 */
object ShuffleHelper {
    @JvmStatic
    fun MutableList<Song>.shuffleAt(current: Int) {
        if (isEmpty()) return

        if (current in 0 until size) {
            val song: Song = removeAt(current)
            shuffle()
            add(0, song)
        } else {
            shuffle()
        }
    }
}
